adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -121,6 +121,121 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
     replaceRdd(tFn(rdd))
   }
 
+  private[rdd] val sortedTrait: SortedTrait
+  private[rdd] def sorted = sortedTrait.sorted
+  private[rdd] def partitionMap = sortedTrait.partitionMap
+
+  /**
+   * This case class is the source of all sorted knowledge handling in ADAM. We
```

Comment:
`sortedTrait` is simply a case class with no methods and only two fields at this point. `sortedTrait` is abstract because we want all the fields to be immutable, so if we factored out the two fields (`sorted` and `partitionMap`) they would have to be abstract too, I believe. That way the child classes can continue to declare these on instantiation because they know whether they are sorted or not.